pip/commands/install.py

```diff
@@ -88,8 +88,21 @@ def __init__(self, *args, **kw):
             dest='upgrade',
             action='store_true',
             help='Upgrade all specified packages to the newest available '
-                 'version. This process is recursive regardless of whether '
-                 'a dependency is already satisfied.'
+                 'version. The handling of dependencies depends on the '
+                 'upgrade-strategy used.'
+        )
+
+        cmd_opts.add_option(
+            '--upgrade-strategy',
+            dest='upgrade_strategy',
+            default='eager',
+            choices=['non-eager', 'eager'],
```

Comment:
You may want to consider whether "non-eager" is clear to a user. The [original proposal](https://github.com/pypa/pip/issues/59#issuecomment-52433585) of "only-if-needed" is not much longer. That's what the pip docs call it now: https://pip.pypa.io/en/latest/user_guide/#only-if-needed-recursive-upgrade